net/isc-dhcp/files/dhcpd.init

```diff
@@ -307,8 +308,34 @@ boot() {
 	start "$@"
 }
 
+add_interface_trigger() {
+	local cfg=$1
+	local trigger ignore
+
+	config_get trigger "$cfg" interface
+	config_get_bool ignore "$cfg" ignore 0
+
+	if [ -n "$trigger" -a $ignore -eq 0 ] ; then
+		procd_add_reload_interface_trigger "$trigger"
+	fi
+}
+
 service_triggers()
 {
-	procd_add_reload_trigger "dhcp"
-	procd_add_raw_trigger "interface.*" 3000 /etc/init.d/dhcpd reload
+	if [ -n "$DHCPD_BOOT" ] ; then
+		# Make the first start robust to slow interfaces; wait a while
+		procd_add_raw_trigger "interface.*.up" 5000 /etc/init.d/dhcpd restart
```

Comment:
Why can't you install immediately interface specific triggers ?
It seems dhcpd is not started if DHCPD_BOOT is set; so a specific trigger should be able to start dhcpd or do I miss something ?

Reply:
@EricLuehrsen If you're good with the current code I'll merge it.

I'm running it and it seems fine.

Reply:
Still I don't understand why you first have to install a non interface specific trigger and later switch to interface specific triggers ...

Reply:
@dedeckeh because until a better solution is made apparent for hardware with weird network delays vs dhcpd expectations, this one seems to work.

@pprindeville yeah, if it works, then merge it. It can suffer through attempts at perfection later.

Reply:
@EricLuehrsen I rest my case as I fail to understand the root cause based on the provided info both in the commit message and my question

Reply:
@nbd168 Can you jump into this for a quick sec and break a tie?

Reply:
@jow- ?

Reply:
Merge at your own discretion - no opinion on this. I agree with @dedeckeh that the trigger switching is unexpected but if its proven to work then fine with me.

Reply:
@dedeckeh what would it look like with a more consistent set of triggers?

Reply:
The trick is the long delay and its effects. This maybe needs 5 or more seconds at boot to avoid the original problem, but this could be too much normally. The correction to that needs a hard "restart" which calls "stop" and "start" and reconfigured triggers and _delay_ in the process. To be kinder later, "reload" is opportunistic and affects as little as possible if it can avoid it. However if you can accept consequences of the long delay at all times, then the following is the short version.
```
service_triggers() {
	# delay all triggers 5 seconds for dhcpd (set procd script tools global)
	PROCD_RELOAD_DELAY=5000
	procd_add_reload_trigger "network" "dhcp"
	config_load dhcp
	config_foreach add_interface_trigger dhcp
}
```

Reply:
I would expect you can use the same approach as in the dnsmasq init script; if bound to an interface and the interface is not up don't start. Next to that install the interface specific trigger in service_triggers and not the interface wildcard trigger which seems to cause the problems (?)
If the interface comes up and due to the interface specific trigger it should be possible to start dhcp. But as @jow said merge it at your own discretion but I find the whole construction very unusual and does not fill me with confidence the root cause of the problem is properly fixed.

Reply:
@dedeckeh the problem I had solving this was that a pure DHCP server should be able to start hard at START=90. It serves on static interfaces. There should be no problem and no procd tricks. On some hardware it needs a long delay. The boot hook with normal 1 second bound interface triggers also fail. The raw restart trigger jams a long delay fix in at boot, but restores typical behavior otherwise.

Reply:
@EricLuehrsen the problem I had solving this was that a pure DHCP server should be able to start hard at START=90.

Strongly disagree on this.  I'm working on glue that also takes the static DHCP assignments and generates DNS/rDNS records and feeds that to bind.  It's conceivable that services at earlier run-levels (25 and later) for instance might need to be able resolve those names to addresses.

In that case, at the very least it should be able to field DNS resolution requests over `lo` and 127.0.0.1 even if no other interfaces are up.

Reply:
It seems that the diff is not about performance improvement, and issue 6282 in the commit message is not relevant.

Sorry, I failed to see the full picture here.  The twist of triggers here does not look right though.  If the current construct does not work, better to fix or create new construct for the situation if needed, not to add hacks here and there.